content/200-concepts/100-components/02-prisma-client/250-tracing.mdx

```diff
@@ -0,0 +1,362 @@
+
+context.setGlobalContextManager(contextManager)
+
+const provider = new BasicTracerProvider({
+  resource: new Resource({
+    [SemanticResourceAttributes.SERVICE_NAME]: 'test-tracing-service',
+    [SemanticResourceAttributes.SERVICE_VERSION]: '1.0.0',
+  }),
+})
+
+// Console exporter
+const consoleExporter = new ConsoleSpanExporter()
+provider.addSpanProcessor(new SimpleSpanProcessor(consoleExporter))
+provider.register()
+
+// Register Prisma Instrumentation
+registerInstrumentations({
+  instrumentations: [new PrismaInstrumentation()],
+})
+
+async function main() {
+  const prisma = new PrismaClient()
+
+  const email = `user.${Date.now()}@prisma.io`
+
+  await prisma.user.create({
+    data: {
+      email: email,
+    },
+  })
+}
+
+main()
+```
+
+## Send tracing to Jaeger
+
+To send tracing to Jaeger, add the following to you application code:
+
+```ts
+import { context } from '@opentelemetry/api'
+import { AsyncHooksContextManager } from '@opentelemetry/context-async-hooks'
+import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-http'
+import { registerInstrumentations } from '@opentelemetry/instrumentation'
+import { Resource } from '@opentelemetry/resources'
+import {
+  BasicTracerProvider,
+  SimpleSpanProcessor,
+} from '@opentelemetry/sdk-trace-base'
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions'
+import { PrismaInstrumentation } from '@prisma/instrumentation'
+import { PrismaClient } from '@prisma/client'
+
+const contextManager = new AsyncHooksContextManager().enable()
+
+context.setGlobalContextManager(contextManager)
+
+const provider = new BasicTracerProvider({
+  resource: new Resource({
+    [SemanticResourceAttributes.SERVICE_NAME]: 'test-tracing-service',
+    [SemanticResourceAttributes.SERVICE_VERSION]: '1.0.0',
+  }),
+})
+
+// Jaeger exporter
+const otlpTraceExporter = new OTLPTraceExporter()
+provider.addSpanProcessor(new SimpleSpanProcessor(otlpTraceExporter))
+provider.register()
+
+// Register Prisma Instrumentation
+registerInstrumentations({
+  instrumentations: [new PrismaInstrumentation()],
+})
+
+async function main() {
+  const prisma = new PrismaClient()
+
+  const email = `user.${Date.now()}@prisma.io`
+
+  await prisma.user.create({
+    data: {
+      email: email,
+    },
+  })
+}
+
+main()
+```
+
+To run Jaeger locally, use the following docker command:
+
+```console
+docker run --rm --name jaeger \
+  -e COLLECTOR_ZIPKIN_HOST_PORT=:9411 \
+  -e COLLECTOR_OTLP_ENABLED=true \
+  -p 6831:6831/udp \
+  -p 6832:6832/udp \
+  -p 5778:5778 \
+  -p 16686:16686 \
+  -p 4317:4317 \
+  -p 4318:4318 \
+  -p 14250:14250 \
+  -p 14268:14268 \
+  -p 14269:14269 \
+  -p 9411:9411 \
+  jaegertracing/all-in-one:1.35
+```
+
+## Reduce performance impact
+
+If your application sends a large number of spans to a collector, this can have a performance impact. You can use the following approaches to reduce this impact.
```

Comment:
Can the performance impact be quantified? How would I figure this out?

Reply:
Good point - I'll add a note about this.